src/cli/hook-utils/get-check-update-data.js

```diff
@@ -0,0 +1,313 @@
+ * @param cwd the current working directory of the CLI project
+ */
+async function getJSONFiles(cwd) {
+  const jsonFiles = ['package.json'];
+  const jsonDepFiles = [];
+  const inaccessibleFiles = [];
+
+  for (const fileName of jsonFiles) {
+    try {
+      const jsonFile = await getJSON(`${cwd}/${fileName}`);
+      const jsonIsParsable =
+        jsonFile &&
+        typeof jsonFile === 'object' &&
+        !Array.isArray(jsonFile) &&
+        jsonFile.dependencies;
+
+      if (jsonIsParsable) {
+        jsonDepFiles.push(fileName);
+      }
+    } catch (err) {
+      inaccessibleFiles.push({ name: fileName, error: err });
+    }
+  }
+  return { jsonDepFiles, inaccessibleFiles };
+}
+
+/**
+ * Pulls dependencies from a given file and JSON structure.
+ * @param json JSON information that includes dependencies
+ * @param fileName name of the file that the dependency list is coming from
+ */
+async function extractDependencies(json, fileName) {
+  // Determine if the JSON passed is an object
+  const jsonIsParsable =
+    json !== null && typeof json === 'object' && !Array.isArray(json);
+
+  if (jsonIsParsable) {
+    const boltCurrentVersionOutput = JSON.parse(await getBoltCurrentVersion());
+    let boltCurrentVersion;
+    if (boltCurrentVersionOutput !== '') {
+      boltCurrentVersion =
+        boltCurrentVersionOutput['dependencies']['@slack/bolt']['version'];
+    }
+    const denoCurrentVersionOutput = JSON.parse(await getDenoCurrentVersion());
+    let denoCurrentVersion;
+    if (denoCurrentVersionOutput !== '') {
+      denoCurrentVersion =
+        denoCurrentVersionOutput['dependencies']['@slack/deno-slack-sdk'][
+          'version'
+        ];
+    }
+
+    return [
+      [
+        '@slack/bolt',
+        {
+          version: boltCurrentVersion,
+        },
+      ],
+      [
+        '@slack/deno-slack-sdk',
+        {
+          version: denoCurrentVersion,
+        },
+      ],
+    ];
+  }
+
+  return [];
+}
+
+/**
+ * Gets the latest module version from NPM.
+ * @param moduleName the module that the latest version is being queried for
+ */
+async function fetchLatestModuleVersion(moduleName) {
+  const res = await fetch(`https://registry.npmjs.org/${moduleName}`, {
+    method: 'GET',
+  })
+    .then((response) => {
+      return response.json();
+    })
+    .catch((error) => {
+      throw new Error(`${moduleName} not found on NPM!`);
+    });
+  if (res['dist-tags']) {
+    if (moduleName === '@slack/deno-slack-sdk') {
+      return res['dist-tags']['latest'];
+    } else if (moduleName === '@slack/bolt') {
+      return res['dist-tags']['next-gen'];
+    }
+  }
+
+  return '';
+}
+
+/**
+ * Checks if a dependency's upgrade from a current to the latest version will cause a
+ * breaking change.
+ * @param current current dependency version in project
+ * @param latest most up-to-date dependency version available on NPM
+ */
+function hasBreakingChange(current, latest) {
```

Comment:
How are we classifying breaking changes' for Bolt JS since we have the beta version? Right now it's just detecting if it's different at all:

<img width="723" alt="Screen Shot 2022-10-06 at 6 25 27 PM" src="https://user-images.githubusercontent.com/12901850/194429756-133c0a0e-6741-4e3c-ad14-11710f3a89cc.png">

Reply:
@hello-ashleyintech - Great question - We don't have a great way of tracking this given the naming schema suggested in the maintainer's guide. I wonder whether it's because beta releases off a feature branch` could all potentially be breaking - there are somewhat less guarantees than when we release off `main`. What do you think of making this return `has breaking change  === true and including a message / url to the associated Release announcement in github where there's more detail on changes?

Reply:
ahh, that makes sense! For the message/URL, are you thinking that would surface on the CLI side where the breaking change messaging surfaces (when running `slack upgrade`)?

Reply:
actually - I see where to pass in a `message` param to the CLI. I'll give that a try 🙌